Miner/Vladimir/MinerGame.cs

```diff
@@ -10,48 +10,205 @@ class MinerGame : IMinerGame
+            public bool openCell;             // признак Открыта/Закрыта клетка
+        }
+
+        public bool OpenStatus (int row, int col)
+        {
+            return poleGame[row][col].openCell;
+        }
+
+        public void SetCanMakeBomb(bool inCanMakeBomb)
+        {
+            if (!IsGameStarted)
+            {
+                canMakeBomb = inCanMakeBomb;
+            }
         }
+
+        public int CountBomb
+        {
+            get { return countBomb; }
+        }
+
+        private int height, width;
+        private CellsGame[][] poleGame;
+
+        public MinerGame(Tuple<int, int> rowsCols)
+        {
+            height = rowsCols.Item1; // height = rowsCols.Item2;                            под старый тест мастера
+            width = rowsCols.Item2; // width = rowsCols.Item1;                              под старый тест мастера
+
+            poleGame = new CellsGame[Height][]; // poleGame = new CellsGame[Width][];       под старый тест мастера
+
+            for (int i = 0; i < Height; i++) // for (int i = 0; i < Width ; i++)            под старый тест мастера
+            {
+                poleGame[i] = new CellsGame[Width]; // poleGame[i] = new CellsGame[Height]; под старый тест мастера
+                for (int j = 0; j < Width; j++)     // for (int j = 0; j < Height; j++)     под старый тест мастера
+                {
+                    poleGame[i][j].yesOrNoBomb = false;
+                    poleGame[i][j].weightCells = 0;
+                    poleGame[i][j].flagOfPlayer = false;
+                    poleGame[i][j].qгestionOfPlayer = false;
+                    poleGame[i][j].openCell = false;
+                }
+            }
+        }
+
+        //------------------------подсчет веса клетки--------------------------------
+        private void WeightCells()
+        {
+            // Создание и инициализация массива с нулевыми первыми и последними строками и солбцами
+            int[][] workArray = new int[poleGame.Length + 2][];
+            int secondIndex = poleGame[0].Length;
+
+            for (int i = 0; i < poleGame.Length + 2; i++)
+            {
+                workArray[i] = new int[secondIndex + 2];
+            }
+
+            // Подсчет весов ячеек входного массива (вес ячейки = сумма количества бомб в окружающих ячейках)
+            // Заливка значений входного массива в массив, в котором значения 1-й, последней строки и столбы = 0
+            for (int i = 0; i < poleGame.Length; i++)
+            {
+                for (int j = 0; j < poleGame[i].Length; j++)
+                {
+                    workArray[i + 1][j + 1] = poleGame[i][j].yesOrNoBomb ? 1 : 0;
+                }
+            }
+            // Посчет весов ячеек входного массива (вес ячейки = количества бомб в окружающих ячейках)
+            int k = 1;
+            for (int i = 0; i < poleGame.Length; i++)
+            {
+                int l = 1;
```

Comment:
what are l and k? please give reasonable names. l at all isn't recommended since it is similar to I and 1.